File: plus-multiply.py
# Online Python compiler (interpreter) to run Python online.
# Write Python 3 code in this online editor and run it.
array1 = [3,8]
array2 = [9,3,9]
print('array1', array1)
print('array2',array2)
print('#'*30)

def plus(array1, array2):
    array1 = array1[::-1]
    array2 = array2[::-1]
    if len(array1) - len(array2) > 0:
        array2+=[0]*(len(array1)-len(array2))
    else:
        array1+=[0]*(len(array2)-len(array1))
    ans = []
    carry = 0
    
    for i in range(max(len(array1), len(array2))):
        subans = array1[i] + array2[i] + carry
        ans.append(subans%10)
        carry = subans//10
    
    if carry!=0:
        ans+=[carry]
    return ans[::-1]
    
print('plusAns:',plus(array1, array2))
print('#'*30)

def multiply(array1, array2):
    array1 = array1[::-1]
    array2 = array2[::-1]
    if len(array1) - len(array2) > 0:
        array2+=[0]*(len(array1)-len(array2))
    else:
        array1+=[0]*(len(array2)-len(array1))
    
    ans = []
    
    for i in range(len(array1)):
        subAns = [0]*i
        subCarry = 0
        for j in range(len(array2)):
            subAns.append((array1[j]*array2[i]+subCarry)%10)
            subCarry = (array1[j]*array2[i]+subCarry)//10
        if subCarry!=0:
            subAns+=[subCarry]
        subAns = subAns[::-1]
        ans = plus(subAns, ans)

    return ans

print('multiplyAns:',multiply(array1, array2))


        
        
        
        
        

File: plus-multiply.ts
let array1: number[] = [1,2,3]
let array2: number[] = [5,6]

function plus(array1: number[], array2: number[]){
    if(array1.length-array2.length>0){
        array2 = new Array(array1.length-array2.length).fill(0).concat(array2)
    } else {
        array1 = new Array(array2.length-array1.length).fill(0).concat(array1)
    }

    let ans: number[] = []
    let carry = 0
    
    for (let i = array1.length-1; i>=0; i--) {
        let subAns = array1[i]+array2[i] + carry
        ans.push(subAns%10)
        carry = Math.floor(subAns/10)
    }

    if(carry!=0){
        ans.push(carry)
    }

    return ans.reverse()
}

console.log(plus(array1, array2))

function multiply(array1: number[], array2: number[]){
    let ans = []

    for (let i = array1.length)
}